### DISCUSSION TITLE
How do I swap text?

### DISCUSSION THREAD

**Discussion by linuxer0x42** [meow-edit/meow#152] [Q&A] [answered]
Hello everyone, been using meow for a few days and trying out some of its features.. 

I've been trying to swap words around with `R`, but it doesn't work, this is what I tried:

I select two words with: 
- `w` + `e` if the other word is in front of current word
- `w` + `b` if the other word is backwards from current word

Then I press `G` for secondary selection, and then I do `e` + `b` or just `b` to create multiple cursors at each word beginning in the backwards direction, and then I press `R` to swap words, it doesn't swap the words...

NVM, fixed, I read the description of the `R` keybind function `meow-swap-grab`;
Description is: "Swap region and secondary selection",
I thought you had to create multiple cursors at each word within a secondary selection, but you have to select a region for one word and a secondary selection for another, I don't fully understand Beacon mode yet, but I am learning...

How would I do this for more than one line though?, like:
line 1: `foo bar`
line 2: `foo bar`

Would I select both lines with `x 2`, then press `G` to create secondary selection, then go to the word `foo` and press `w` to create normal selection, and then press `R` to swap `foo` with `bar`?? It isn't working? How do I do this for more than one cursor?

**Comment by DogLooksGood** [comment, marked as answer]
Hi
In your case, 
- select 2 line, press `G`, activate secondary selection
- start kmacro recording with `F3`
- select `bar` with `w`
- `G` to create secondary selection
- select `foo` with `b b w`
- `R` to swap
- `F4` for finish kmacro recording, then kmacro will be applied to all beacons.

**Reply by linuxer0x42** [reply]
Oh nice, thank you! This is exactly what I wanted 
Kmacros are very cool

**Reply by DogLooksGood** [reply]
https://www.youtube.com/watch?v=HhuAlAcWWeU

And I have a short video for this, sorry for my bad english.  Video is good for demonstrate the ideas, but it's very hard for me to make a good video tutorial.

**Reply by linuxer0x42** [reply]
I will watch it very soon, thanks for making the video even if you have bad English, it can help beginners like me to understand meow better!
